test/shared/src/main/scala/zio/test/mock/internal/ProxyFactory.scala

```diff
@@ -164,11 +172,18 @@ object ProxyFactory {
                         if (updatedChild.saturated) resetTree(updatedChild)
                         else updatedChild
 
+                      def inRange(value: Int): Boolean =
+                        if (range.end != -1) range contains value
+                        else {
+                          val fakeUnboundedRange = range.start to Int.MaxValue by range.step
```

Comment:
@ self => we could use Long internally instead of Int for the counter and have a formula like `(value - range.start) % range.step == 0` to check if its satisfied, therefore moveing the limit from Int.MaxValue to Long.MaxValue for open-ended repetition upper bounds, however letting the test repeat `Int.MaxValue` times is already an overkill, I can't imagine ATM a scenario where a higher limit is useful